src/overlays/actors/ovl_En_Niw_Lady/z_en_niw_lady.c

```diff
@@ -20,43 +36,536 @@ const ActorInit En_Niw_Lady_InitVars = {
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ this->message_end_code   ☆☆ %d\n" VT_RST, this->unk_262);
+        osSyncPrintf("\n\n");
+        if (Text_GetFaceReaction(globalCtx, 8) == 0) {
+            if (this->actor.textId == 0x503C) {
+                func_80078884(NA_SE_SY_ERROR);
+                this->unk_26C = 2;
+                this->unk_262 = 5;
+                this->actionFunc = func_80ABA654;
+                return;
+            }
+            this->unk_26E = phi_s1 + 1;
+            if (phi_s1 == 7) {
+                func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+                this->unk_26C = 1;
+                this->unk_262 = 5;
+                this->unk_26A = this->cuccosInPen;
+                osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ 柵内BIT変更前 ☆☆ %x\n" VT_RST, gSaveContext.infTable[0x19]);
+                gSaveContext.infTable[0x19] &= 0x1FF;
+                osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ 柵内BIT変更後 ☆☆ %x\n" VT_RST, gSaveContext.infTable[0x19]);
+                osSyncPrintf("\n\n");
+                this->actionFunc = func_80ABA654;
+                return;
+            }
+            if (this->unk_26A != this->cuccosInPen) {
+                if (this->cuccosInPen < this->unk_26A) {
+                    func_80078884(NA_SE_SY_ERROR);
+                } else if (phi_s1 + 1 < 9) {
+                    func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+                }
+            }
+            if (this->unk_26A < this->cuccosInPen) {
+                this->unk_26A = this->cuccosInPen;
+                return;
+            }
+        }
+    } else {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    }
+}
+
+void func_80ABA654(EnNiwLady* this, GlobalContext* globalCtx) {
+    if (this->unk_262 == func_8010BDBC(&globalCtx->msgCtx) && func_80106BC8(globalCtx) != 0) {
+        func_80106CCC(globalCtx);
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ ハート ☆☆☆☆☆ %d\n" VT_RST, this->unk_26C);
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 爆弾   ☆☆☆☆☆ %d\n" VT_RST, this->unk_272);
+        osSyncPrintf("\n\n");
+        this->unk_26E = 0xB;
+        if (!(gSaveContext.itemGetInf[0] & 0x1000)) {
+            this->actor.parent = NULL;
+            this->unk_284 = 0xF;
+            func_8002F434(&this->actor, globalCtx, 0xF, 100.0f, 50.0f);
+            this->actionFunc = func_80ABAC00;
+            return;
+        }
+        if (this->unk_26C == 1) {
+            this->unk_284 = 0x55;
+            func_8002F434(&this->actor, globalCtx, 0x55, 100.0f, 50.0f);
+            this->actionFunc = func_80ABAC00;
+        }
+        this->actionFunc = func_80ABA244;
+    }
+}
+
+static s16 sTradeItemTextIds[] = { 0x503E, 0x503F, 0x5047, 0x5040, 0x5042, 0x5043,
+                                   0x5044, 0x00CF, 0x5045, 0x5042, 0x5027, 0x0000 };
+
+void func_80ABA778(EnNiwLady* this, GlobalContext* globalCtx) {
+    //☆☆☆☆☆ Adult message check ☆☆☆☆☆
+    osSyncPrintf("\x1b[32m☆☆☆☆☆ アダルトメッセージチェック ☆☆☆☆☆ \n\x1b[m", this);
```

Comment:
VT macros should be used here. Also no `this` argument.
```suggestion
    osSyncPrintf("\x1b[32m☆☆☆☆☆ アダルトメッセージチェック ☆☆☆☆☆ \n\x1b[m");
```